### PR TITLE
[ramda] Bump types-ramda package to latest 

### PR DESCRIPTION
**Pull request by Harris-Miller**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by Harris-Miller** [comment]
`types-ramda`'s `package.json` includes [this](https://github.com/ramda/types/pull/115/files) in the latest version:
```
  "typesVersions": {
    "<5": {
      "*": [
        "ts4/*"
      ]
    }
  },
```
It works as expected, as we can see from the now-failing tests. Those tests are failing because they're typed to pass for ts@5. How do I write separate tests for <5 ? Or can I mute those tests? Annotate them that they for >=5 only?

**Comment by jakebailey** [comment]
If the package were fully on DT, you'd have types versions here and separate tests, and that's the only way to achieve separate versions testing.

Theoretically you can just do that, copy the package into a ts4.9 dir for that and below and make it pass.

But I understand that may be a little weird given it's duplication just for testing. Are you already running tests in your repo? Do you need the coverage here?

File: types/ramda/tsconfig.json [hunk 1]
```diff
@@ -154,7 +154,6 @@
         "test/nthArg-tests.ts",
         "test/o-tests.ts",
         "test/objOf-tests.ts",
-        "test/omit-tests.ts",
```
Comment (jakebailey) [inline types/ramda/tsconfig.json:157]: This file was modified, but dropped from the array; not sure if that's intentional but it'll mean the tests won't get run. (We don't currently check for unreferenced files, unfortunately)

**Comment by Harris-Miller** [comment]
I decided to add `as const` to the falling tests. the ts5 versions of Pick/Omit use `const` in the generic def, so adding `const` if what you'd want to keep the consistent behavior. So it makes sense to do

**Review by jakebailey — APPROVED** [review]

**Comment by Harris-Miller** [comment]
Ready to merge

**Comment by Harris-Miller** [comment]
Thanks @jakebailey for your timely review of this one! The only thing updated in `types-ramda` was the missing `typesVersions {}` prop in `package.json` which I left out in a release a month ago, this update unblocks a lot of `ramda` users upgrading to `0.29.1` now!

**Pull request merged**